### ISSUE TITLE
Consider streamlining BERT code?

### ISSUE THREAD

**Issue by fierceX** [dmlc/gluon-nlp#986]
## Description
Have you considered simplifying the BERT model code? I think the BERT model code is too complicated now.
Now the BERT main model code uses `attention_cell.py`, `seq2seq_encoder_decoder.py`, `transformer.py`, `bert.py`, a total of four files, which I think is unreasonable.
For example: `BERTEncoderCel`l inherits from `BaseTransformerEncoderCel`l, but only the default parameters are different.` BERTEncoder` inherits from `BaseTransformerEncoder`, but only the default parameters are different. `BERTLayerNorm` is inherited from `nn.LayerNorm`, but only the default parameters are different.
`BERTPositionwiseFFN` inherits from `BasePositionwiseFFN`, but only the default parameters are different.
Added a `use_bert` parameter to the parameter, but only to use some default parameters, so why not set some default parameters directly. The bert-related code is placed in `bert.py`, and the base class is stored in the `transformer.py` file, but when use_bert=True, there is a need to import some subclass code with default parameters from `bert.py`. In addition:
The `BaseTransformerEncoder` base class also inherits from` Seq2SeqEncoder`, but `Seq2SeqEncoder` does not have additional code. The `_get_attention_cell` function I think should be in `utils.py` or `attention_cell.py` instead of `seq2seq_encoder_decoder.py`.
@dmlc/gluon-nlp-committers 

## References
- list reference and related literature
- list known implementations

**Comment by szha** [comment]
@eric-haibin-lin @leezu

**Comment by leezu** [comment]
Yes, we would like to refactor it. Would you like to work on it @fierceX?

**Comment by eric-haibin-lin** [comment]
I agree. The code is now complicated and not copy-paste-able. The inheritance of BaseTransformerEncoder and BaseTransformerEncoderCell makes it hard to modify the code. We should just have separate implementations for different models

**Comment by leezu** [comment]
I did some basic refactoring today @fierceX, but you're welcome to improve further on it. Will ping you in the PR.
